src/parser.pegjs

```diff
@@ -50,20 +50,26 @@
   function buildList(head, tail, index) {
     return [head].concat(extractList(tail, index));
   }
-}
+}}
 
 // ---- Syntactic Grammar -----
 
 Grammar
-  = __ initializer:(Initializer __)? rules:(Rule __)+ {
+  = __ topLevelInitializer:(TopLevelInitializer __)? initializer:(Initializer __)? rules:(Rule __)+ {
       return {
         type: "grammar",
+        topLevelInitializer: extractOptional(topLevelInitializer, 0),
         initializer: extractOptional(initializer, 0),
         rules: extractList(rules, 0),
         location: location()
       };
     }
 
+TopLevelInitializer
+  = "{" code:CodeBlock "}" EOS {
```

Comment:
hm.  are there code blocks that could match this for parse-level initializers that happen to start with `{` and end with `}`

Reply:
Technically you could have has many curly braces as you'd want so, yes, you could have a per-parse initializer that has double curly braces... but it serves no purpose whatsoever and would be very weird code formatting.

The only situation it would arise is if someone wanted to scope the entire code and do so without indentation... something like:

```js
{{
  let scoped;
  // some code
}}
```

But:
1. I expect such code to be indented
2. I also expect unscoped code to exist after or before the scoped code

Something like:

```js
{
  {
    let scoped;
    // some code
  }
  // other code
}
```

The likelihood of a double curly braced initializer in the wild seems extremely low but, yes, it _could_ happen.

Reply:
nod.  and there's a work-around if it does break something of putting a `{{}}` in front if you really want.